### PR TITLE
Test view

### PR DESCRIPTION
**Pull request by farouqdaif1**
Use Capybara to write integration tests for each view in your project.

Login page:

I can see the username and password inputs and the "Submit" button.
When I click the submit button without filling in the username and the password, I get a detailed error.
When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.
User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    it 'I can see the profile picture for each user' do
+      expect(page.find('img')['src']).to have_content @user.photo
+    end
+
+    it 'number of posts of the user' do
+      expect(page).to have_content 'Number of posts: 1'
+    end
+  end
+end
```
Comment (ahmetbozaci) [inline spec/features/user_index_spec.rb]: - [x] Please add a test when the user clicks on the user redirect to the user's show page. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Integration_specs_project.md#:~:text=When%20I%20click%20on%20a%20user%2C%20I%20am%20redirected%20to%20that%20user%27s%20show%20page.)

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      expect(page).to have_content 'Like'
+      expect(page).to have_content 'Delete'
+    end
+
+    it "When I click to see all posts, it redirects me to the user's post's index page." do
+      click_on 'See all posts'
+      expect(current_path).to eq user_posts_path user_id: User.last.id
+    end
+  end
+end
```
Comment (ahmetbozaci) [inline spec/features/user_show_spec.rb]: - [x] You should add tests to see the user's first 3 posts.
- [x] You should add a test that redirects to the post's show page when you click on a user's post. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Integration_specs_project.md#:~:text=I%20can%20see%20a%20button,to%20that%20post%27s%20show%20page.)

File: spec/features/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    end
+
+    it 'When I click a user post, it redirects me to that post s show page.' do
+      visit user_post_path(@user.id, @post.id)
+      expect(page).to have_content 'Comment'
+      expect(page).to have_content 'Like'
+      expect(page).to have_content 'Delete'
+    end
+  end
+end
```
Comment (ahmetbozaci) [inline spec/features/post_index_spec.rb]: - [x] You should add a test for the user's picture. 
- [x] You should add test for pagination if there are more posts than fit on the view.
[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Integration_specs_project.md#:~:text=I%20can%20see%20a%20section%20for%20pagination%20if%20there%20are%20more%20posts%20than%20fit%20on%20the%20view.)

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @farouqdaif1  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] Please fix the request test errors.


image

___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @farouqdaif1,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
- [ ] Good job adding the other missing tests. However, some of your request tests are still failing and one is also pending. Kindly resolve this. Your route paths seem to be inaccessible within the request spec files and you are nesting `describe` blocks within another `describe` block.

  failing-tests

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — APPROVED** [review]
Hi @farouqdaif1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**